### PR TITLE
fix(a11y): basic forms demo

### LINKED ISSUE #2085: Fix a11y issues in Core Form Demo
Related to: https://github.com/patternfly/patternfly-org/issues/1346

These a11y issues are coming from the Core demo for Basic forms

- Checkbox inputs with the same name attribute value must be part of a group
- Headings must not be empty

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #2085 

- Added headingtext to the forms demo 
- made the `name=...` assignments from the `check-input` unique as this is what we have in the check component. This fixes the error for "Checkbox inputs with the same name attribute value must be part of a group"... I am wondering though @jgiardino do these checks need a form fieldset around them like in the select component?

### CODE REVIEW

File: src/patternfly/demos/BasicForms/examples/index.js [hunk 1]
```diff
@@ -11,12 +11,13 @@ import docs from '../docs/code.md';
 
 export const Docs = docs;
 
-export default (props) => {
+export default props => {
   const simpleFormDemoExample = SimpleFormDemoExample();
   const horizontalFormDemoExample = HorizontalFormDemoExample();
+  const headingText = 'Basic Forms Demo';
```
Comment (mcoker) [inline src/patternfly/demos/BasicForms/examples/index.js]: should be sentence case

```suggestion
  const headingText = 'Basic forms demo';
```

**Comment by redallen** [comment]
Merging master to get latest CircleCI changes.

File: src/patternfly/demos/BasicForms/examples/index.js (hunk 1 above)
Comment (redallen) [reply src/patternfly/demos/BasicForms/examples/index.js]: Done.

**Pull request merged**